client/state/stats/lists/selectors.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import get from 'lodash/get';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import i18n from 'lib/mixins/i18n';
+import {
+	getSerializedStatsQuery
+} from './utils';
+
+/**
+ * Returns true if currently requesting stats for the statType and query combo, or false
+ * otherwise.
+ *
+ * @param  {Object}  state    Global state tree
+ * @param  {Number}  siteId   Site ID
+ * @param  {String}  statType Type of stat
+ * @param  {Object}  query    Stats query object
+ * @return {Boolean}          Whether stats are being requested
+ */
+export function isRequestingSiteStatsForQuery( state, siteId, statType, query ) {
+	const serializedQuery = getSerializedStatsQuery( query );
+	return !! get( state.stats.lists.requesting, [ siteId, statType, serializedQuery ] );
+}
+
+/**
+ * Returns object of stats data for the statType and query combo, or null if no stats have been
+ * received.
+ *
+ * @param  {Object}  state    Global state tree
+ * @param  {Number}  siteId   Site ID
+ * @param  {String}  statType Type of stat
+ * @param  {Object}  query    Stats query object
+ * @return {?Object}           Data for the query
+ */
+export function getSiteStatsForQuery( state, siteId, statType, query ) {
+	const serializedQuery = getSerializedStatsQuery( query );
+	return get( state.stats.lists.items, [ siteId, statType, serializedQuery ], null );
+}
+
+/**
+ * Returns a parsed object of statsStreak data for a given query, or default "empty" object
+ * if no statsStreak data has been received for that site.
+ *
+ * @param  {Object}  state    Global state tree
+ * @param  {Number}  siteId   Site ID
+ * @param  {Object}  query    Stats query object
+ * @return {Object}           Parsed Data for the query
+ */
+export const getParsedStreakDataForQuery = createSelector(
+	( state, siteId, query ) => {
+		const response = { days: {}, best: { day: 0, percent: 0 }, max: 0 };
```

Comment:
I wonder if this would be a good opportunity to split the data we're constructing here into more granular selectors that can be composed together or used in more specific circumstances, e.g. `getPostsStatsPerWeekday`, `getBestStatsWeekday`, `getPostsStatsWeekdayPercent`, `getStatsPostsWeekdayMax`. Might also help readability of the code in splitting this into well-described functions.